src/main/java/edu/kit/hci/soli/service/UserService.java

```diff
@@ -40,4 +42,18 @@ public  UserService(UserRepository userRepository) {
     public boolean isAdmin(User user) {
         return user.getUserId().equals("admin");
     }
+
+    public @NotNull User resolveGuestUser(String email) {
+        String id = "guest/" + email;
```

Comment:
You are not using the marker? Or am I misreading what is the marker supposed to do?

Reply:
The marker is added to the form as a password, since we need something to fill that field.
As such, it only ever reaches the `SoliUserDetailsService`